File: Lib/Telerik/Examples/QuickStart/SplitContainer/FirstLook/LayoutProperties.vb
Imports System.ComponentModel
Imports System.Text
Imports Telerik.WinControls.UI
Imports Telerik.WinControls.Enumerations
Imports Telerik.WinControls.Primitives
Imports Telerik.WinControls
Imports Telerik.QuickStart.WinControls

Namespace Telerik.Examples.WinControls.SplitContainer.FirstLook
	Partial Public Class LayoutProperties
		Inherits ExamplesRadForm
		Private previouslySelectedNode As RadTreeNode = Nothing

		Public Sub New()
			InitializeComponent()

			AddHandler FormClosing, AddressOf AdvancedLayoutDesiger_FormClosing
			AddHandler radioHorizontal.ToggleStateChanged, AddressOf radioHorizontal_ToggleStateChanged
			AddHandler radioVertical.ToggleStateChanged, AddressOf radioVertical_ToggleStateChanged
			AddHandler checkSplitterFixed.ToggleStateChanged, AddressOf checkSplitterFixed_ToggleStateChanged
			AddHandler checkFixed.ToggleStateChanged, AddressOf checkFixed_ToggleStateChanged
			AddHandler checkCollapsed.ToggleStateChanged, AddressOf checkCollapsed_ToggleStateChanged
		End Sub

		Private Sub checkCollapsed_ToggleStateChanged(ByVal sender As Object, ByVal args As StateChangedEventArgs)
			CType(Me.radTreeView1.SelectedNode.Tag, SplitPanel).Collapsed = Me.checkCollapsed.ToggleState = ToggleState.On

			Dim splitContNode As RadTreeNode = Me.radTreeView1.SelectedNode.Parent

			Dim splitContainer As Telerik.WinControls.UI.RadSplitContainer = CType(splitContNode.Tag, Telerik.WinControls.UI.RadSplitContainer)

			Dim canRemoveNode As Boolean = True

			Do While canRemoveNode
				If TypeOf splitContNode.Nodes(splitContNode.Nodes.Count - 1).Tag Is SplitterElement Then
					splitContNode.Nodes.RemoveAt(splitContNode.Nodes.Count - 1)
				Else
					canRemoveNode = False
				End If
			Loop

			For i As Integer = 0 To splitContainer.Splitters.Count - 1
				Dim node As New RadTreeNode(String.Format("Splitter{0}", i + 1))
				node.Tag = splitContainer.Splitters(i)
				splitContNode.Nodes.Add(node)
			Next i
		End Sub

		Private Sub checkFixed_ToggleStateChanged(ByVal sender As Object, ByVal args As StateChangedEventArgs)
			If args.ToggleState = Telerik.WinControls.Enumerations.ToggleState.On Then
				CType(Me.radTreeView1.SelectedNode.Tag, SplitPanel).SizeInfo.SizeMode = Telerik.WinControls.UI.Docking.SplitPanelSizeMode.Absolute
			Else
				CType(Me.radTreeView1.SelectedNode.Tag, SplitPanel).SizeInfo.SizeMode = Telerik.WinControls.UI.Docking.SplitPanelSizeMode.Auto
			End If
		End Sub

		Private Sub checkSplitterFixed_ToggleStateChanged(ByVal sender As Object, ByVal args As StateChangedEventArgs)
			CType(Me.radTreeView1.SelectedNode.Tag, SplitterElement).Fixed = Me.checkSplitterFixed.ToggleState = ToggleState.On
		End Sub

		Private Sub radioVertical_ToggleStateChanged(ByVal sender As Object, ByVal args As StateChangedEventArgs)
			If args.ToggleState = ToggleState.On Then
				CType(Me.radTreeView1.SelectedNode.Tag, Telerik.WinControls.UI.RadSplitContainer).Orientation = Orientation.Vertical
			End If
		End Sub

		Private Sub radioHorizontal_ToggleStateChanged(ByVal sender As Object, ByVal args As StateChangedEventArgs)
			If args.ToggleState = ToggleState.On Then
				CType(Me.radTreeView1.SelectedNode.Tag, Telerik.WinControls.UI.RadSplitContainer).Orientation = Orientation.Horizontal
			End If
		End Sub

		Private Sub AdvancedLayoutDesiger_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs)
			SetDefaultColors()
		End Sub

		Protected Overrides Sub OnLoad(ByVal e As EventArgs)
			MyBase.OnLoad(e)

			AddHandler radTreeView1.SelectedNodeChanged, AddressOf radTreeView1_SelectedNodeChanged

			Me.radTreeView1.TreeViewElement.DrawBorder = True

			SetTreeNodeReferences()

		End Sub

		Private Sub SetTreeNodeReferences()
			Dim ownerForm As Telerik.Examples.WinControls.SplitContainer.FirstLook.Form1 = CType(Me.Owner, Telerik.Examples.WinControls.SplitContainer.FirstLook.Form1)

			Me.radTreeView1.Nodes(0).Tag = ownerForm.RadSplitContainer1
			Me.radTreeView1.Nodes(0).Nodes(0).Tag = ownerForm.SplitPanel1
			Me.radTreeView1.Nodes(0).Nodes(1).Tag = ownerForm.SplitPanel2
			Me.radTreeView1.Nodes(0).Nodes(2).Tag = ownerForm.SplitPanel3

			For i As Integer = 0 To ownerForm.RadSplitContainer1.Splitters.Count - 1
				Dim node As New RadTreeNode(String.Format("Splitter{0}", i + 1))
				node.Tag = ownerForm.RadSplitContainer1.Splitters(i)
				Me.radTreeView1.Nodes(0).Nodes.Add(node)
			Next i

			Me.radTreeView1.Nodes(0).Nodes(2).Nodes(0).Tag = ownerForm.RadSplitContainer2
			Me.radTreeView1.Nodes(0).Nodes(2).Nodes(0).Nodes(0).Tag = ownerForm.SplitPanel4
			Me.radTreeView1.Nodes(0).Nodes(2).Nodes(0).Nodes(1).Tag = ownerForm.SplitPanel5

			For i As Integer = 0 To ownerForm.RadSplitContainer2.Splitters.Count - 1
				Dim node As New RadTreeNode(String.Format("Splitter{0}", i + 1))
				node.Tag = ownerForm.RadSplitContainer2.Splitters(i)
				Me.radTreeView1.Nodes(0).Nodes(2).Nodes(0).Nodes.Add(node)
			Next i
		End Sub

		Private Sub radTreeView1_SelectedNodeChanged(ByVal sender As Object, ByVal e As RadTreeViewEventArgs)
			Dim type As String = e.Node.Tag.ToString()
			Select Case type
				Case "Telerik.WinControls.UI.RadSplitContainer"
					LoadSplitContainerProperties(CType(e.Node.Tag, Telerik.WinControls.UI.RadSplitContainer))
					splitContainerLayout.BringToFront()
				Case "Telerik.WinControls.UI.SplitterElement"
					LoadSplitterProperties(CType(e.Node.Tag, SplitterElement))
					splitterLayout.BringToFront()
				Case "Telerik.WinControls.UI.SplitPanel"
					LoadSplitPanelProperties(CType(e.Node.Tag, SplitPanel))
					splitPanelLayout.BringToFront()
			End Select

			SetDefaultColors()

			previouslySelectedNode = e.Node
		End Sub

		Private Sub SetDefaultColors()
			If previouslySelectedNode IsNot Nothing Then
				Dim previousType As String = previouslySelectedNode.Tag.ToString()
				Select Case previousType
					Case "Telerik.WinControls.UI.RadSplitContainer"
						CType((CType((CType(previouslySelectedNode.Tag, Telerik.WinControls.UI.RadSplitContainer)).Parent, RadPanel)).PanelElement.Children(1), BorderPrimitive).ResetValue(VisualElement.ForeColorProperty, ValueResetFlags.Local)
					Case "Telerik.WinControls.UI.SplitterElement"
						Dim fillP As FillPrimitive = CType((CType(previouslySelectedNode.Tag, SplitterElement)).Children(0), FillPrimitive)
						fillP.ResetValue(VisualElement.BackColorProperty, ValueResetFlags.Local)
						fillP.ResetValue(RadElement.VisibilityProperty, ValueResetFlags.Local)
						fillP.ResetValue(FillPrimitive.GradientStyleProperty, ValueResetFlags.Local)
						fillP.ResetValue(RadElement.ShouldPaintProperty, ValueResetFlags.Local)
					Case "Telerik.WinControls.UI.SplitPanel"
						CType(previouslySelectedNode.Tag, SplitPanel).SplitPanelElement.Border.ResetValue(VisualElement.ForeColorProperty, ValueResetFlags.Local)
				End Select
			End If
		End Sub

		Private Sub LoadSplitPanelProperties(ByVal panel As SplitPanel)
			Me.checkFixed.ToggleState = If(panel.SizeInfo.SizeMode = Telerik.WinControls.UI.Docking.SplitPanelSizeMode.Absolute, ToggleState.On, ToggleState.Off)
			Me.checkCollapsed.ToggleState = If(panel.Collapsed, ToggleState.On, ToggleState.Off)

			panel.SplitPanelElement.Border.ForeColor = Color.Red
		End Sub

		Private Sub LoadSplitterProperties(ByVal splitter As SplitterElement)
			Me.checkSplitterFixed.ToggleState = If(splitter.Fixed, ToggleState.On, ToggleState.Off)

			Dim fillP As FillPrimitive = CType(splitter.Children(0), FillPrimitive)
			fillP.BackColor = Color.Red
			fillP.GradientStyle = Telerik.WinControls.GradientStyles.Solid
			fillP.Visibility = ElementVisibility.Visible
			fillP.ShouldPaint = True
		End Sub

		Private Sub LoadSplitContainerProperties(ByVal splitContainer As Telerik.WinControls.UI.RadSplitContainer)
			If splitContainer.Orientation = Orientation.Horizontal Then
				Me.radioHorizontal.ToggleState = ToggleState.On
			Else
				Me.radioVertical.ToggleState = ToggleState.On
			End If

			CType((CType(splitContainer.Parent, RadPanel)).PanelElement.Children(1), BorderPrimitive).ForeColor = Color.Red
		End Sub
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/SplitContainer/FirstLook/LayoutProperties.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls.UI;
using Telerik.WinControls.Enumerations;
using Telerik.WinControls.Primitives;
using Telerik.WinControls;
using Telerik.QuickStart.WinControls;

namespace Telerik.Examples.WinControls.SplitContainer.FirstLook
{
    public partial class LayoutProperties : ExamplesRadForm
    {
        RadTreeNode previouslySelectedNode = null;

        public LayoutProperties()
        {
            InitializeComponent();

            this.FormClosing += new FormClosingEventHandler(AdvancedLayoutDesiger_FormClosing);
            this.radioHorizontal.ToggleStateChanged += new StateChangedEventHandler(radioHorizontal_ToggleStateChanged);
            this.radioVertical.ToggleStateChanged += new StateChangedEventHandler(radioVertical_ToggleStateChanged);
            this.checkSplitterFixed.ToggleStateChanged += new StateChangedEventHandler(checkSplitterFixed_ToggleStateChanged);
            this.checkFixed.ToggleStateChanged += new StateChangedEventHandler(checkFixed_ToggleStateChanged);
            this.checkCollapsed.ToggleStateChanged += new StateChangedEventHandler(checkCollapsed_ToggleStateChanged);
        }

        void checkCollapsed_ToggleStateChanged(object sender, StateChangedEventArgs args)
        {
            ((SplitPanel)this.radTreeView1.SelectedNode.Tag).Collapsed = this.checkCollapsed.ToggleState == ToggleState.On;

            RadTreeNode splitContNode = this.radTreeView1.SelectedNode.Parent;

            Telerik.WinControls.UI.RadSplitContainer splitContainer = (Telerik.WinControls.UI.RadSplitContainer)splitContNode.Tag;

            bool canRemoveNode = true;

            while (canRemoveNode)
            {
                if (splitContNode.Nodes[splitContNode.Nodes.Count - 1].Tag is SplitterElement)
                {
                    splitContNode.Nodes.RemoveAt(splitContNode.Nodes.Count - 1);
                }
                else
                {
                    canRemoveNode = false;
                }
            }

            for (int i = 0; i < splitContainer.Splitters.Count; i++)
            {
                RadTreeNode node = new RadTreeNode(string.Format("Splitter{0}", i + 1));
                node.Tag = splitContainer.Splitters[i];
                splitContNode.Nodes.Add(node);
            }
        }

        void checkFixed_ToggleStateChanged(object sender, StateChangedEventArgs args)
        {
            if (args.ToggleState == Telerik.WinControls.Enumerations.ToggleState.On)
            {
                ((SplitPanel)this.radTreeView1.SelectedNode.Tag).SizeInfo.SizeMode = Telerik.WinControls.UI.Docking.SplitPanelSizeMode.Absolute;
            }
            else
            {
                ((SplitPanel)this.radTreeView1.SelectedNode.Tag).SizeInfo.SizeMode = Telerik.WinControls.UI.Docking.SplitPanelSizeMode.Auto;
            }
        }

        void checkSplitterFixed_ToggleStateChanged(object sender, StateChangedEventArgs args)
        {
            ((SplitterElement)this.radTreeView1.SelectedNode.Tag).Fixed = this.checkSplitterFixed.ToggleState == ToggleState.On;
        }

        void radioVertical_ToggleStateChanged(object sender, StateChangedEventArgs args)
        {
            if (args.ToggleState == ToggleState.On)
            {
                ((Telerik.WinControls.UI.RadSplitContainer)this.radTreeView1.SelectedNode.Tag).Orientation = Orientation.Vertical;
            }
        }

        void radioHorizontal_ToggleStateChanged(object sender, StateChangedEventArgs args)
        {
            if (args.ToggleState == ToggleState.On)
            {
                ((Telerik.WinControls.UI.RadSplitContainer)this.radTreeView1.SelectedNode.Tag).Orientation = Orientation.Horizontal;
            }
        }

        void AdvancedLayoutDesiger_FormClosing(object sender, FormClosingEventArgs e)
        {
            SetDefaultColors();
        }

        protected override void OnLoad(EventArgs e)
        {
            base.OnLoad(e);

            this.radTreeView1.SelectedNodeChanged += new RadTreeView.RadTreeViewEventHandler(radTreeView1_SelectedNodeChanged);

            this.radTreeView1.TreeViewElement.DrawBorder = true;

            SetTreeNodeReferences();

        }

        private void SetTreeNodeReferences()
        {
            Telerik.Examples.WinControls.SplitContainer.FirstLook.Form1 ownerForm = (Telerik.Examples.WinControls.SplitContainer.FirstLook.Form1)this.Owner;

            this.radTreeView1.Nodes[0].Tag = ownerForm.RadSplitContainer1;
            this.radTreeView1.Nodes[0].Nodes[0].Tag = ownerForm.SplitPanel1;
            this.radTreeView1.Nodes[0].Nodes[1].Tag = ownerForm.SplitPanel2;
            this.radTreeView1.Nodes[0].Nodes[2].Tag = ownerForm.SplitPanel3;

            for (int i = 0; i < ownerForm.RadSplitContainer1.Splitters.Count; i++)
            {
                RadTreeNode node = new RadTreeNode(string.Format("Splitter{0}", i + 1));
                node.Tag = ownerForm.RadSplitContainer1.Splitters[i];
                this.radTreeView1.Nodes[0].Nodes.Add(node);
            }

            this.radTreeView1.Nodes[0].Nodes[2].Nodes[0].Tag = ownerForm.RadSplitContainer2;
            this.radTreeView1.Nodes[0].Nodes[2].Nodes[0].Nodes[0].Tag = ownerForm.SplitPanel4;
            this.radTreeView1.Nodes[0].Nodes[2].Nodes[0].Nodes[1].Tag = ownerForm.SplitPanel5;

            for (int i = 0; i < ownerForm.RadSplitContainer2.Splitters.Count; i++)
            {
                RadTreeNode node = new RadTreeNode(string.Format("Splitter{0}", i + 1));
                node.Tag = ownerForm.RadSplitContainer2.Splitters[i];
                this.radTreeView1.Nodes[0].Nodes[2].Nodes[0].Nodes.Add(node);
            }
        }

        void radTreeView1_SelectedNodeChanged(object sender, RadTreeViewEventArgs e)
        {
            string type = e.Node.Tag.ToString();
            switch (type)
            {
                case "Telerik.WinControls.UI.RadSplitContainer":
                    LoadSplitContainerProperties((Telerik.WinControls.UI.RadSplitContainer)e.Node.Tag);
                    splitContainerLayout.BringToFront();
                    break;
                case "Telerik.WinControls.UI.SplitterElement":
                    LoadSplitterProperties((SplitterElement)e.Node.Tag);
                    splitterLayout.BringToFront();
                    break;
                case "Telerik.WinControls.UI.SplitPanel":
                    LoadSplitPanelProperties((SplitPanel)e.Node.Tag);
                    splitPanelLayout.BringToFront();
                    break;
            }

            SetDefaultColors();

            previouslySelectedNode = e.Node;
        }

        private void SetDefaultColors()
        {
            if (previouslySelectedNode != null)
            {
                string previousType = previouslySelectedNode.Tag.ToString();
                switch (previousType)
                {
                    case "Telerik.WinControls.UI.RadSplitContainer":
                        ((BorderPrimitive)((RadPanel)((Telerik.WinControls.UI.RadSplitContainer)previouslySelectedNode.Tag).Parent).PanelElement.Children[1]).ResetValue(VisualElement.ForeColorProperty, ValueResetFlags.Local);
                        break;
                    case "Telerik.WinControls.UI.SplitterElement":
                        FillPrimitive fillP = (FillPrimitive)((SplitterElement)previouslySelectedNode.Tag).Children[0];
                        fillP.ResetValue(VisualElement.BackColorProperty, ValueResetFlags.Local);
                        fillP.ResetValue(RadElement.VisibilityProperty, ValueResetFlags.Local);
                        fillP.ResetValue(FillPrimitive.GradientStyleProperty, ValueResetFlags.Local);
                        fillP.ResetValue(RadElement.ShouldPaintProperty, ValueResetFlags.Local);
                        break;
                    case "Telerik.WinControls.UI.SplitPanel":
                        ((SplitPanel)previouslySelectedNode.Tag).SplitPanelElement.Border.ResetValue(VisualElement.ForeColorProperty, ValueResetFlags.Local);
                        break;
                }
            }
        }

        private void LoadSplitPanelProperties(SplitPanel panel)
        {
            this.checkFixed.ToggleState = panel.SizeInfo.SizeMode == Telerik.WinControls.UI.Docking.SplitPanelSizeMode.Absolute ? ToggleState.On : ToggleState.Off;
            this.checkCollapsed.ToggleState = panel.Collapsed ? ToggleState.On : ToggleState.Off;

            panel.SplitPanelElement.Border.ForeColor = Color.Red;
        }

        private void LoadSplitterProperties(SplitterElement splitter)
        {
            this.checkSplitterFixed.ToggleState = splitter.Fixed ? ToggleState.On : ToggleState.Off;

            FillPrimitive fillP = (FillPrimitive)splitter.Children[0];
            fillP.BackColor = Color.Red;
            fillP.GradientStyle = Telerik.WinControls.GradientStyles.Solid;
            fillP.Visibility = ElementVisibility.Visible;
            fillP.ShouldPaint = true;
        }

        private void LoadSplitContainerProperties(Telerik.WinControls.UI.RadSplitContainer splitContainer)
        {
            if (splitContainer.Orientation == Orientation.Horizontal)
            {
                this.radioHorizontal.ToggleState = ToggleState.On;
            }
            else
            {
                this.radioVertical.ToggleState = ToggleState.On;
            }

            ((BorderPrimitive)((RadPanel)splitContainer.Parent).PanelElement.Children[1]).ForeColor = Color.Red;
        }
    }
}
